src/perf/parse-all-benchmark.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+import {Analyzer} from '../analyzer';
+import {FSUrlLoader} from '../url-loader/fs-url-loader';
+import {UrlLoader} from '../url-loader/url-loader';
+
+import {Measurement} from './telemetry';
+
+class PermissiveUrlLoader implements UrlLoader {
+  private _realLoader: UrlLoader;
+  constructor(realLoader: UrlLoader) {
+    this._realLoader = realLoader;
+  }
+  canLoad() {
+    return true;
+  }
+  async load(path: string) {
+    try {
+      return await this._realLoader.load(path);
+    } catch (_) {
+      // muddle on!
+    }
+    return '';
+  }
+}
+
+const bowerDir = path.resolve(__dirname, `../../bower_components`);
+const analyzer = new Analyzer(
+    {urlLoader: new PermissiveUrlLoader(new FSUrlLoader(bowerDir))});
+
+const filesToAnalyze: string[] = [];
+
+for (const baseDir of fs.readdirSync(bowerDir)) {
+  const bowerJsonPath = path.join(bowerDir, baseDir, 'bower.json');
+  let bowerJson: any;
+  try {
+    bowerJson = JSON.parse(fs.readFileSync(bowerJsonPath, 'utf-8'));
+  } catch (e) {
+    continue;
+  }
+  const main: string|string[] = bowerJson.main || [];
+  const mains = Array.isArray(main) ? main : [main];
+  for (const mainFile of mains) {
+    if (existsSync(path.join(bowerDir, baseDir, mainFile))) {
+      filesToAnalyze.push(path.join(baseDir, mainFile));
+    }
+  }
+}
+
+
+function existsSync(fn: string): boolean {
+  try {
+    fs.statSync(fn);
+    return true;
+  } catch (_) {
+    return false;
+  }
+}
+
+const fakeFileContents =
+    filesToAnalyze.map(fn => `<link rel="import" href="${fn}">`).join('\n');
+
+function padLeft(str: string, num: number): string {
+  if (str.length < num) {
+    return padLeft(' ' + str, num);
+  }
+  return str;
+}
+
+const measure = async() => {
```

Comment:
hm, I thought clang-format was misformatting it as a declaration. looks like that's not the case though.

converted to declaration